onnxruntime/core/platform/windows/env.cc

```diff
@@ -208,6 +208,12 @@ class WindowsEnv : public Env {
     }
     if (ret.empty())
       return generate_vector_of_n(std::thread::hardware_concurrency());
+
+    /* move first affinity to the tail */
+    size_t first = ret[0];
+    ret.erase(ret.begin());
+    ret.push_back(first);
```

Comment:
why are we doing this?
It is because we prefer to leave core #0 to be free?
The approach has a few uncertainties:
1. The main thread is not necesarilly running on core #0, meaning this calibration might not helping much?
2. Logically we prefer to have thread 0 stay close to thread 1, and thread 2 close to thread 3 since during task parallelism we tend to allocate "cache-close" tasks sequentially to thread 1,2,3,4...